packages/backend/src/controllers/seasons.controller.ts

```diff
@@ -34,16 +34,25 @@ export class SeasonsController {
     return SeasonDto.fromEntity(season!);
   }
 
+  @ApiResponse({
+    status: 200,
+    description: "Returns season by id",
+    type: SeasonDto,
+  })
+  @Get("/:id")
+  async getSeasonById(@Param("id") id: string): Promise<SeasonDto> {
+    const season = await this.seasonsService.getSeason(id);
+    return SeasonDto.fromEntity(season!);
+  }
+
   @ApiResponse({
     status: 200,
     description: "Returns rules for season",
     type: BoardConfigDto,
   })
-  @Get("/rules/:id")
-  async getCurrentSeasonRules(
-    @Param("id") id: string,
-  ): Promise<BoardConfigDto> {
-    const season = await this.seasonsService.getCurrentSeason();
+  @Get("/:id/rules")
+  async getSeasonRules(@Param("id") id: string): Promise<BoardConfigDto> {
+    const season = await this.seasonsService.getSeason(id);
```

Comment:
same here

Reply:
maybe it should be thrown in the service right away instead? then we can change the return type from the `getSeason` call to `ISeason` instead of `ISeason | null`

Reply:
implemented a try using the same logic as in the getOffSeasonService. 

```
  public async getSeason(seasonId: string): Promise<ISeason> {
    const season = await this.repo.getById(seasonId, true);
    if (!season) {
      throw new Error("Season does not exist");
    }
    return season;
  }
```